boa3_test/tests/cli_tests/test_compile.py

```diff
@@ -0,0 +1,160 @@
+from boa3_test.tests.boa_test import BoaTest  # needs to be the first import to avoid circular imports
+
+import io
+import os
+from boa3.cli import main
+from boa3.internal import constants, env
+from boa3.internal.neo3.vm import VMState
+from boa3_test.test_drive.testrunner.neo_test_runner import NeoTestRunner
+from contextlib import redirect_stdout
+from unittest.mock import patch
+
+
+class TestCli(BoaTest):
+    default_folder = 'test_cli/smart_contract_folder'
+
+    @patch('sys.argv', ['neo3-boa', '-h'])
+    def test_cli_help(self):
+        with redirect_stdout(io.StringIO()) as output, self.assertRaises(SystemExit):
+            main()
+        cli_output = output.getvalue()
+
+        self.assertIn('usage: neo3-boa [-h] [-v] {compile}', cli_output)
+        self.assertIn(f'neo3-boa by COZ - version {constants.BOA_VERSION}', cli_output)
+        self.assertIn('Write smart contracts for Neo3 in Python', cli_output)
+
+    @patch('sys.argv', ['neo3-boa', '--version'])
+    def test_cli_version(self):
+        with redirect_stdout(io.StringIO()) as output, self.assertRaises(SystemExit):
+            main()
+        cli_output = output.getvalue()
+
+        self.assertIn(f'neo3-boa {constants.BOA_VERSION}', cli_output)
+
+    @patch('sys.argv', [
+        'neo3-boa',
+        'compile',
+        constants.PATH_SEPARATOR.join(
+            [env.PROJECT_ROOT_DIRECTORY, 'boa3_test', 'test_cli', 'smart_contract_folder', 'smart_contract.py']
+        )
+    ])
+    def test_cli_compile(self):
+        sc_name = 'smart_contract.py'
+        nef_path, manifest_path = self.get_deploy_file_paths(sc_name)
+        debug_info_path = nef_path.replace('nef', 'nefdbgnfo')
+
+        if os.path.isfile(nef_path):
+            os.remove(nef_path)
+        if os.path.isfile(manifest_path):
+            os.remove(manifest_path)
+        if os.path.isfile(debug_info_path):
+            os.remove(debug_info_path)
```

Comment:
Is there a reason to remove these files? Does the test fail if the files exist?

Reply:
It was to guarantee that the CLI command would generate the files if something went wrong with the compilation